### DISCUSSION TITLE
Platform dependency

### DISCUSSION THREAD

**Discussion by AntoineD** [tox-dev/tox#2071] [Ideas]
With tox 3, if a setting of a testenv depends on the platform (say windows), it seems that the only solution to declare this *specialized* env is to add a factor (say win). In my projects I thus have to differentiate how the envs are executed
- `tox -e py38` on linux and macos
- `tox -e py38-win` on windows

To be clear, executing `tox -e py38` on windows cannot work.

This applies to most of the envs and is quite cumbersome. Would it be possible to have conditional settings that do not depend on factor, at least for the platform?

**Comment by jugmac00** [comment]
Does this answer help you?

https://stackoverflow.com/a/66904046/672833

**Reply by AntoineD** [reply]
Thanks for trying to help, unfortunately this is already what I do.

**Reply by jugmac00** [reply]
I am not entirely sure what problem you encounter.

You have environments which only work on specific platforms. At one point you need to tell tox which commands / envs work on which platform.

You could also create two different `tox.ini` files - e.g. one for Windows, one for Linux.

Maybe you could rephrase your question? Otherwise at least I cannot help you.

**Comment by AntoineD** [comment]
OK I'll try to rephrase.

I'd like to run tox on windows just like on the other platforms: `tox -e py38` for all platforms, but because some settings differ for windows, I have to use a factor.

Is there a way, with tox 3 or for the future tox 4, to have conditional settings that are not factors?

**Reply by gaborbernat** [reply]
The flake8 example @jugmac00 linked has no factorials included 🤔 and works.

**Reply by AntoineD** [reply]
From the [related doc](https://tox.readthedocs.io/en/latest/example/platform.html):
- isn't `mywindows` a factor?
- should `tox -e py36` execute a command? (It does not for me)

**Reply by AntoineD** [reply]
Does it work for you?
